src/common/libkvs/kvs_lookup.c

```diff
@@ -47,6 +68,12 @@ flux_future_t *flux_kvs_lookupat (flux_t *h, int flags, const char *key,
     flux_future_t *f;
     json_t *obj = NULL;
 
+    if (!h || !key || strlen (key) == 0) {
+        errno = EINVAL;
+        return NULL;
+    }
+    if (validate_lookup_flags (flags) < 0)
+        return NULL;
     if (!treeobj) {
```

Comment:
just an idle suggestion -- a couple lines of code could be removed by combining the two conditionals here:
```C
    if (!h || !key || strlen (key) == 0 || validate_lookup_flags (flags) < 0) {
        errno = EINVAL;
        return NULL;
    }
```

as a bonus `validate_lookup_flags` doesn't have to set its own errno (it is internal and only fails in one way anyhow)